backend.go

```diff
@@ -15,12 +18,42 @@ func NewFilesystemBackend(root, tmp string) (storage.Backend, error) {
 		return nil, err
 	}
 
+	storage, err := findAllBackends(fsobj, packs, root)
+	if err != nil {
+		return nil, err
+	}
+
 	return &filesystemBackend{
 		fs:       fsobj,
-		backends: []storage.Storage{fsobj, packs},
+		backends: storage,
 	}, nil
 }
 
+func findAllBackends(mainLoose *fileStorer, mainPacked *pack.Storage, root string) ([]storage.Storage, error) {
+	storage := make([]storage.Storage, 2)
+	storage[0] = mainLoose
+	storage[1] = mainPacked
+	f, err := os.Open(path.Join(root, "info", "alternates"))
```

Comment:
Do we need to keep this file open for any reason? If not, let's add a `defer f.Close()` after handling the error, like so:

```go
f, err := os.Open(...)
if err != nil {
        // ...
}
defer f.Close()
```